### PR TITLE
Docusaurus web docs

### PR DESCRIPTION
**Pull request by solid-daniilmarchenko**
Example of deployed instance: https://solid-daniilmarchenko.github.io/solid_lints/

### CODE REVIEW

File: doc/docusaurus/docs/Lints Documentation/avoid_late_keyword.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+A `late` keyword rule which forbids using it to avoid runtime exceptions.
```
Comment (yurii-prykhodko-solid) [inline doc/docusaurus/docs/Lints Documentation/avoid_late_keyword.md:1]: Let's not keep generated files in VCS if we can help it.

File: lib/utils/docs_parser/parse_docusaurus_docs.dart [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import 'dart:io';
+
+import 'package:path/path.dart';
+import 'package:solid_lints/utils/docs_parser/output_formatters/docusaurus_formatter.dart';
+import 'package:solid_lints/utils/docs_parser/parsers/docs_parser.dart';
+
+void main() async => DocsParser(
```
Comment (yurii-prykhodko-solid) [inline lib/utils/docs_parser/parse_docusaurus_docs.dart:7]: Let's move all executable scripts under `/bin`.

File: lib/utils/docs_parser/parse_docusaurus_docs.dart [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+void main() async => DocsParser(
+      formatter: DocusaurusFormatter(),
+    ).parse(
+      Directory(
+        normalize(
+          join(Directory.current.path, 'lib', 'lints'),
+        ),
+      ),
+    );
```
Comment (yurii-prykhodko-solid) [inline lib/utils/docs_parser/parse_docusaurus_docs.dart:15]: Let's add input-output parameters properly. For now we seem to be hardcoding a good lot, and it's really not obvious where things come from and where they go -- will make it tricky to figure out once we get back to this in some time.

Let's add some simple parameters that would make it apparent what the script takes in and what it outputs.

File: .github/workflows/gh-pages-test-deploy.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
+name: Test deployment
```
Comment (yurii-prykhodko-solid) [inline .github/workflows/gh-pages-test-deploy.yaml:1]: Is this needed in `master`?

File: bin/parse_web_docs.dart [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      mandatory: true,
+    )
+    ..addOption(
+      'docs-dir',
+      abbr: 'o',
+      help: 'Parser output path. i.e "docusaurus/docs" directory',
+      defaultsTo: docusaurusDefaultPath,
+    )
+    ..addOption(
+      'output-dir',
```
Comment (yurii-prykhodko-solid) [inline bin/parse_web_docs.dart:31]:
```suggestion
      'output-dirname',
```

Comment (yurii-prykhodko-solid) [reply bin/parse_web_docs.dart:31]: Scratch it -- just remove the argument and make the path part of the `docs-dir` argument.

File: bin/parse_web_docs.dart [hunk 6]
```diff
@@ -0,0 +1,73 @@
+import 'dart:io';
```
Comment (yurii-prykhodko-solid) [inline bin/parse_web_docs.dart:1]: rename to `generate_doc_templates.dart`

File: .github/workflows/gh-pages-test-deploy.yaml (hunk 4 above)
Comment (solid-daniilmarchenko) [reply .github/workflows/gh-pages-test-deploy.yaml:1]: yep, this checks if documentation could be successfully built

File: bin/parse_web_docs.dart (hunk 5 above)
Comment (solid-daniilmarchenko) [reply bin/parse_web_docs.dart]: it's necessary for separating outer folder which contains `intro.md` file and the inner folder with lints documenation

Comment (solid-daniilmarchenko) [reply bin/parse_web_docs.dart]: but, after thinking a little, i think we could resolve path for `intro.md` file by just using parent directory of output dir. 
will change it tomorrow

**Pull request merged**